src/modules/Analysis.ts

```diff
@@ -1306,4 +1329,57 @@ export default class Analysis extends VolumeIndicatorAnalysis {
       });
     });
   }
+
+  getPattern(trend: string){
+    const trends: object[] = trendMap[trend].map((pattern: string) => {
+      return this[pattern]().then((r: any) => {
+        return r
+      }).catch((e: any) => {
+        console.dir(e, { depth: null });
+      });
+    })
+
+    return Promise.all(
+      trends
+    ).then((patterns: any) => {
+      let data: any = {}
+      trendMap[trend].map((key: string, index: number) => {
+        data[key] = patterns[index].length
+      })
+      data['total'] = Object.values(data).reduce((a:any, b:any) => a + b);
```

Comment:
avoid use of any... alos, I don't see the the logic for checking how recent the patter was detected.

Reply:
> avoid the use of any

Yes, it is a draft as I could reason the data type.

> I don't see the the logic for checking how recent the patter was detected

Recent pattern? we are checking the last X candle then run the analysis on it.

Please clarify if I miss something.

Reply:
`CandlePattern[]`